internal/core/textlayout/sharedparley.rs

```diff
@@ -962,20 +964,71 @@ fn parse_markdown(string: &str) -> RichText<'_> {
                     let end = paragraph.text.len();
                     paragraph.formatting.push(FormattedSpan { range: start..end, style });
+                } else {
+                    let mut expecting_color_attribute = false;
+
+                    for token in htmlparser::Tokenizer::from(&*html) {
+                        match token {
+                            Ok(htmlparser::Token::ElementStart { local: tag_type, .. }) => {
+                                match &*tag_type {
+                                    "u" => {
+                                        style_stack.push((
+                                            Style::Underline,
+                                            rich_text.paragraphs.last().unwrap().text.len(),
+                                        ));
+                                    }
+                                    "font" => {
+                                        expecting_color_attribute = true;
+                                    }
+                                    _ => {
+                                        unimplemented!("Unhandled start tag type: {}", tag_type)
+                                    }
+                                }
+                            }
+                            Ok(htmlparser::Token::Attribute {
+                                local: key,
+                                value: Some(value),
+                                ..
+                            }) => match &*key {
+                                "color" => {
+                                    assert!(expecting_color_attribute);
+                                    expecting_color_attribute = false;
+
+                                    let value =
+                                        i_slint_common::color_parsing::parse_color_literal(&*value)
+                                            .or_else(|| {
+                                                i_slint_common::color_parsing::named_colors()
+                                                    .get(&*value)
+                                                    .copied()
+                                            })
+                                            .expect("invalid color value");
+
+                                    style_stack.push((
+                                        Style::Color(crate::Color::from_argb_encoded(value)),
+                                        rich_text.paragraphs.last().unwrap().text.len(),
+                                    ));
+                                }
+                                _ => {
+                                    std::unimplemented!("attribute {}: {}", key, value);
+                                }
+                            },
+                            Ok(htmlparser::Token::ElementEnd { .. }) => {}
+                            _ => {
+                                std::unimplemented!("{:?}", token);
```

Comment:
this should also be a warning, same for all the other asserts and debug_assert.

In fact, this function should ideally return as well a list of error that would be shown as compilation error if we do the parsing at compile time, or print as warning if done at runtime.